src/components/Identicon/StatusIcon.tsx

```diff
@@ -34,5 +69,10 @@ export default function StatusIcon({ connectionType, size }: { connectionType: C
       break
   }
 
-  return <IconWrapper size={size ?? 16}>{image}</IconWrapper>
+  return (
+    <IconWrapper size={size ?? 16}>
+      {hasSocks && isNavbarEnabled && <Socks />}
```

Comment:
Refactor this to first check for `isNavBarEnabled`

    {isNavBarEnabled ?
      (
        {hasSocks && <Socks />}
        <Identicon />
      ) : image
    }